src/picongpu/CMakeLists.txt

```diff
@@ -254,6 +266,24 @@ if( (Boost_VERSION EQUAL 105500) AND
       "${CUDA_NVCC_FLAGS} \"-DBOOST_NOINLINE=__attribute__((noinline))\" ")
 endif()
 
+# PMacc (ab)uses boost::result_of for non-functors. This define forces boost to use the
+# result<> member template of the target type
+if(Boost_VERSION LESS 105500)
+    add_definitions(-DBOOST_RESULT_OF_USE_TR1)
+else()
+    # Boost 1.55 adds support for another define that makes result_of look for the result<> template
+    # and falls back to decltype if none is found. This is great for the transition from the "wrong"
+    # usage to the "correct" one as both can be used. But:
+    # 1) Cannot be used in 7.0 due to nvcc bug:
+    #    http://stackoverflow.com/questions/31940457/make-nvcc-output-traces-on-compile-error
```

Comment:
hm, the author of SO link I posted got the same problem from [boost in this repo](https://github.com/dealii/dealii/issues/1323#issuecomment-130562072), posted a non-boost example [in SO](https://stackoverflow.com/questions/31969644/compilation-error-with-nvcc-and-c11-need-minimal-failing-example) that generally shows that problem with `decltype` in CUDA 7.0 and added in the original github issue the boost work-around `BOOST_RESULT_OF_USE_TR1` for reference.

If you mention the `boost version` and the work-around via  that is missing in your SO we can keep the link, but pls update the SO for it, it is lacking the solution and affected versions.